### PR TITLE
Add the ability to cancel a stopwatch span.

### PR DESCRIPTION
**Pull request by stefan-vainberg**
Added a new method `cancel` to StopWatch that allows the current span to be removed from the stopwatch stack without being reported.

### CODE REVIEW

**Comment by twk** [comment]
@nipunn1313 -- sorry, I clicked your name by accident.

File: stopwatch.py [hunk 1]
```diff
@@ -184,6 +184,10 @@ def _reset(self):
         self._root_annotations = []
         self._slow_annotations = {}
 
+        # Dictionary of span names that have been cancelled in the current
+        # context. Used to ensure that a cancelled span is not redundantly ended as well.
+        self._cancelled_spans = {}
```
Comment (nipunn1313) [inline stopwatch.py]: Can we make this a set? Looks like the value is not used?

File: stopwatch.py [hunk 2]
```diff
@@ -206,7 +210,10 @@ def timer(self, name, bucket=None, start_time=None, end_time=None):
             self.add_annotation('Exception', type(e).__name__, event_time=end_time)
             raise
         finally:
-            self.end(name, end_time=end_time, bucket=bucket)
+            if name in self._cancelled_spans:
+                del self._cancelled_spans[name]
+            else:
+                self.end(name, end_time=end_time, bucket=bucket)
```
Comment (nipunn1313) [inline stopwatch.py]: Can we roll this check into self.end(). Feels like we should always check cancelled_spans on end(), not just within a context manager.

File: stopwatch.py [hunk 3]
```diff
@@ -346,6 +351,24 @@ def format_last_report(self):
     # Private methods
     #################
 
+    def _pop_stack(self, name, end_type='end'):
```
Comment (nipunn1313) [inline stopwatch.py]: Can you update this docstring to include the arguments. It looks like the end_type argument is only used in the assertion message? Can we just re-check the cancelled_spans to figure out the assertion message and avoid the extra arg?

File: test_stopwatch.py [hunk 4]
```diff
@@ -47,6 +47,29 @@ def test_default_exports(self):
...
+        """Test that spans can be cancelled while inside a span context."""
+        sw = StopWatch()
+        with sw.timer('root'):
+            with sw.timer('child'):
+                sw.cancel('child')
+                with sw.timer('grand'):
+                    pass
+        agg_values = sw.get_last_aggregated_report().aggregated_values
+        assert len(agg_values) == 2
+        assert all([span in agg_values for span in ('root', 'root#grand')])
```
Comment (nipunn1313) [inline test_stopwatch.py]: Can you add a test which will create a second root span? I think we have to clear the cancelled spans set at some point?

Comment (nipunn1313) [inline test_stopwatch.py]: It looks like there isn't a normal test which creates multiple root spans, but I think this is a behavior we expect to work. I believe the client/server already uses it that way.

Might want to add a test for that as well. Something along these lines (with asertions / boilerplate added.
```
def test_multiple_roots(self):
    with sw.timer('root'):
        pass
    with sw.timer('root'):
        passs
```

**Pull request merged**